### PR TITLE
feat: Voice chat now disabled by default. Added allowlist for voice chat

### PR DESCRIPTION
**Pull request by pablitar**
There are now four controls for enabling / disabling voice chat:
- `voiceChatEnabled` flag in meta configuration -> enables/disables voice chat globally
- `voiceChatAllowlist` list in meta configuration -> Enables voice chat for those included in the list by address (0xabcdef01234...)
- `VOICE_CHAT_ENABLED` in the URL overrides this and enables it for this session
- `VOICE_CHAT_DISABLED` in the URL overrides this and enables it for this session

### CODE REVIEW

File: kernel/packages/entryPoints/unity.ts [hunk 1]
```diff
@@ -58,26 +69,22 @@ initializeUnity(container)
...
     i.ConfigureHUDElement(HUDElementID.NFT_INFO_DIALOG, { active: true, visible: false })
     i.ConfigureHUDElement(HUDElementID.TELEPORT_DIALOG, { active: true, visible: false })
-    i.ConfigureHUDElement(HUDElementID.CONTROLS_HUD, { active: true, visible: false })
-    i.ConfigureHUDElement(HUDElementID.EXPLORE_HUD, { active: true, visible: false })
-    i.ConfigureHUDElement(HUDElementID.HELP_AND_SUPPORT_HUD, { active: true, visible: false })
-    i.ConfigureHUDElement(HUDElementID.USERS_AROUND_LIST_HUD, { active: voiceChatEnabled, visible: false })
+
+    //NOTE(Brian): Scene download manager uses meta config to determine which empty parcels we want
+    //             so ensuring meta configuration is initialized in this stage is a must
+    //NOTE(Pablo): We also need meta configuration to know if we need to enable voice chat
```
Comment (eordano) [inline kernel/packages/entryPoints/unity.ts:78]: Please let's stop adding these notes with `@authors` -- it's very prone to aging badly.

File: kernel/packages/shared/meta/selectors.ts [hunk 2]
```diff
@@ -34,4 +34,9 @@ export const getMessageOfTheDay = (store: RootMetaState): MessageOfTheDayConfig
   store.meta.config.world ? store.meta.config.world.messageOfTheDay || null : null
 
 export const isVoiceChatEnabled = (store: RootMetaState): boolean =>
-  !getCommsConfig(store).voiceChatDisabled && !VOICE_CHAT_DISABLED_FLAG
+  (!!getCommsConfig(store).voiceChatEnabled && !VOICE_CHAT_DISABLED_FLAG) || VOICE_CHAT_ENABLED_FLAG
+
+export const getVoiceChatWhitelist = (store: RootMetaState): string[] => getCommsConfig(store).voiceChatAllowlist ?? []
```
Comment (eordano) [inline kernel/packages/shared/meta/selectors.ts]:
```suggestion
export const getVoiceChatAllowlist = (store: RootMetaState): string[] => getCommsConfig(store).voiceChatAllowlist ?? []
```

File: kernel/packages/shared/meta/selectors.ts [hunk 3]
```diff
@@ -34,4 +34,9 @@ export const getMessageOfTheDay = (store: RootMetaState): MessageOfTheDayConfig
...
+
+export const isVoiceChatEnabledFor = (store: RootMetaState, userId: string): boolean =>
+  isVoiceChatEnabled(store) || getVoiceChatWhitelist(store).includes(userId)
```
Comment (eordano) [inline kernel/packages/shared/meta/selectors.ts]:
```suggestion
  isVoiceChatEnabled(store) || getVoiceChatAllowlist(store).includes(userId)
```

**Review by caminoezequiel — APPROVED** [review]

File: kernel/packages/entryPoints/unity.ts [hunk 4]
```diff
@@ -39,6 +40,16 @@ const observer = renderStateObservable.add((isRunning) => {
   }
 })
 
+function configureTaskbar(i: UnityInterface, voiceChatEnabled: boolean) {
```
Comment (pbosio) [inline kernel/packages/entryPoints/unity.ts]: Not a big fan of this function name, but something more descriptive like`configureTaskbarRelatedHUDs` or `configureTaskbarDependentHUDs` seems too long. Can we at least add a comment there... cause it's not that clear that all those HUDs have a dependency on the Taskbar. 
Also if we can move the configuration of `USERS_AROUND_LIST_HUD` (it's the only one independent from the taskbar) outside this function, like only calling it if the `getCurrentIdentity` succeeded right below the  `configureTaskbar`, that would be great (insert meme here)

**Review by pbosio — APPROVED** [review]

**Pull request merged**